core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -32,22 +32,57 @@
  */
 public abstract class RetryingClient<I extends Request, O extends Response>
         extends SimpleDecoratingClient<I, O> {
+
+    private static final int FALLBACK_DEFAULT_MAX_RETRY_ATTEMPTS = 5;
+    private static final int DEFAULT_MAX_RETRY_ATTEMPTS;
+
+    static {
+        String value = System.getProperty("com.linecorp.armeria.defaultMaxRetryAttempts", "5");
+        if (value.matches("^-?\\d+$")) { // to check whether integer or not.
```

Comment:
If the value is <= 0, then it would throw an exception when the AttemptLimitingBackoff is created. But I think it needs a validation check as you mentioned.
About the checking whether integer of not part, you are right about that.
Because most of the parameters would be an integer value not other values.
Thank you~!!